src/main/java/de/doubleslash/keeptime/view/ReportController.java

```diff
@@ -194,6 +201,25 @@ protected void updateItem(final TableRow item, final boolean empty) {
       rootItem.setExpanded(true);
    }
 
+   private void collapseExpandTable(){
+
+      pressed=!pressed;
+
+      if(pressed){
+         //CollapseAll
+         for (int i=0; i<rootItem.getChildren().size(); i++){
+            rootItem.getChildren().get(i).setExpanded(false);
+         }
+         expandColapseButton.setText("Expand");
```

Comment:
it is not clear to me what the state is when 'pressed' is true. It seems pressed true means the items are not expanded. the variable should be renamed so the relation to expanded/collapsed is clear(er).

Also it is not clear what the initial state is of the items - as they do not use any of the variables newly introduced. E.g. if I start the app everything is expanded, if I press the button it may want to expand again? The state should be in sync - best by using the same variables representing the state.